### DISCUSSION TITLE
Hiding the Dock

### DISCUSSION THREAD

**Discussion by woomia** [helloSystem/hello#246] [General]
Can you add an option that allows the dock to be hidden/autohidden? It takes up a chunk of real estate I could use.

**Comment by probonopd** [comment]
This is not implemented yet. We are not sure whether the Dock is here to stay or whether it will be replaced with something like this.

![image](https://user-images.githubusercontent.com/2480569/140637749-275f06f0-57e8-4a32-81b8-a9eda24db011.png)

https://github.com/helloSystem/Menu/issues/21

**Reply by woomia** [reply]
I like the dock a lot, I just wish it could autohide.

**Reply by probonopd** [reply]
I see. I don't know how to implement it and I am lacking the time to research it at the moment. If you have some time, maybe you can find out?

Probably there are some existing docks written in Qt or ideally Qml that can do this, can you identify the relevant source code for the autohiding?

**Reply by woomia** [reply]
I can do that, or try. I can't code but I'm pretty good at reading code. The Mac OS X dock (the early versions), could autohide the dock, and you could move it to the right or left.

**Reply by woomia** [reply]
ksmoothdock has a lot of KDE depends BUT this where the autohide code lives:
https://github.com/dangvd/ksmoothdock/blob/master/src/view/dock_panel.cc
https://github.com/dangvd/ksmoothdock/blob/master/src/view/dock_panel.h

**Reply by probonopd** [reply]
Cool. Can you open a feature request in https://github.com/cyberos/cyber-dock/issues/new please regarding auto-hide, mentioning these links? Our Dock is based on that project's Dock. Thanks for your help!

**Reply by woomia** [reply]
You got it!

**Reply by alphamodh0** [reply]
> This is not implemented yet. We are not sure whether the Dock is here to stay or whether it will be replaced with something like this.
> 
> ![image](https://user-images.githubusercontent.com/2480569/140637749-275f06f0-57e8-4a32-81b8-a9eda24db011.png)
> 
> [helloSystem/Menu#21](https://github.com/helloSystem/Menu/issues/21)

In my opinion, the **Dock is perfect**, it only needs some adjustments and functionalities. According to me (**and I think most**) removing the dock would be a mistake.

**Reply by kettle-7** [reply]
e x t r a         c l i c k s

**Comment by woomia** [comment]
Okay, I figured where the config file for cyber-dock hides, it's called dock.conf and Its in /home/user/.config/cyberos
it looks like this:
[General]
Direction=1
IconSize=54
Visibility=1

Creating a file in helloSystem doesn't work, its either ignored, or causes the system to freeze but might be a good place to start if you want to enable this feature.

**Comment by probonopd** [comment]
We would probably need to port over the helloSystem specific changes to the newest version of https://github.com/cutefishos/dock, or port their changes into our version (but not the Cutefish OS specific dependencies).

If anyone has some time, please consider sending a pull request.

**Reply by kettle-7** [reply]
I might, I'm in covid quarantine :~(

On the plus side I have more time without work...

**Reply by woomia** [reply]
That would be awesome!!

**Reply by probonopd** [reply]
Best wishes to you @kettle-7.
Yes, this would be great.

**Reply by kettle-7** [reply]
I have done a dock like this, but in wayland and not designed for x but maybe I can port it.

**Reply by probonopd** [reply]
Thanks. Please note that we need autohide to be ported into the helloSystem dock, not some other dock.
Because the helloSystem dock has integration with application directories.

**Reply by kettle-7** [reply]
Yes, fortunately this was for something I tried with the helloSystem desktop on wayland. It didn't work out very well but I got the appdir integration well.

**Comment by woomia** [comment]
I noticed this when trying to using another dock, its just not the same.

**Comment by woomia** [comment]
I'm definitely in the "keep the dock" crowd, I just want to hide it.

**Comment by probonopd** [comment]
Someone would need to port the CutefishOS Dock https://github.com/helloSystem/Dock/issues/24.
